src/storage/storage.ts

```diff
@@ -86,10 +86,17 @@ export class Storage {
 
     const bucketData: Parameters<Database['createBucket']>[0] = data
 
-    if (data.fileSizeLimit) {
+    if (typeof data.fileSizeLimit === 'number' || typeof data.fileSizeLimit === 'string') {
       bucketData.file_size_limit = await this.validateMaxSizeLimit(data.fileSizeLimit)
```

Comment:
should probably be named as parseMaxSizeLimit since it also does the string to number conversion or the parse and validate should be separated out